PT-727 Publish the version of the service in the status object

ydrisR opened the pull request:
Modify the common object "Status" to add the version of the service.
Also add unit tests to prevent regressions

ydrisR committed:
Add unit tests to cover the existing functions of the status package

ydrisR committed:
Factorize code to prepare the inclusion of the version

ydrisR committed:
Add a basic and simple version package to display the application version number

Inspired, but extremely simplified,  from the platform project.

ydrisR committed:
Add the versioning info in the status object

CedricPerotto commented on clients/status/status.go:
```diff
@@ -3,41 +3,41 @@ package status
+	Reason  string `json:"reason"`
+	Version string `json:"version"`
 }
 
```
If we don't want to embed the Version in all status struct (error status for ex.). We could Extend the Status struct only for API status.

i.e. 
```
type StatusApi struct{
	Status 
	Version string `json:"version"`
}
```
with the factory function NewStatusApi 
```
func NewStatusApi(status Status) StatusApi {
	s := StatusApi{Status : status, Version: version.Long()}
	return s
}
```
The NewStatusApi  would be only called for api.GetStatus calls.

ydrisR replied:
👍

ydrisR committed:
review the version package architecture

based on peer review comments

ydrisR committed:
implement a singleton and it associated tests

ydrisR committed:
update the status object to use the new version singleton.

Create a separate apiStatus object to distinguish from the legacy status which is used in all request error responses.

ydrisR committed:
correct typo

BigFab commented on clients/version/version.go:
```diff
@@ -0,0 +1,71 @@
+// Variables to be injected at build time
+var (
+	ReleaseNumber string //Release number. i.e. 1.2.3
+	ShortCommit   string
```
```suggestion
	ShortCommit   string //Short commit id. i.e. 048a8d4
```

BigFab commented on clients/version/version.go:
```diff
@@ -0,0 +1,71 @@
+)
+
+func GetVersion() version {
+	once.Do(func() {
```
```suggestion
        //Ensure this is called once so version instance is created as a singleton
	once.Do(func() {
```

BigFab commented on clients/version/version.go:
```diff
@@ -0,0 +1,71 @@
+var instance version
+var once sync.Once
+
+// Variables to be injected at build time
```
```suggestion
// Variables to be injected at build time
// E.g. go build -ldflags "-X $GO_COMMON_PATH/clients/version.ReleaseNumber=$VERSION"
```

BigFab reviewed (commented):
LGTM although I would like to see some more comments that I suggested inline

ydrisR committed:
Add more comments

Co-Authored-By: Fabrice Sougey <33152621+BigFab@users.noreply.github.com>

BigFab reviewed (approved):
LGTM now :)

ydrisR committed:
Update changelog

ydrisR committed:
Merge branch 'feature/PT-727_Enhance_Status_Report' of github.com:mdblp/go-common into feature/PT-727_Enhance_Status_Report
